include/Parser/include/config.h

```diff
@@ -0,0 +1,135 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include "parser.h"
+
+class Config {
+public:
+    // Structs to hold different configuration settings
+    struct Server {
+        std::string servername = "DefaultServer"; // Default server name
+        std::string serverdisplayname = "DefaultServerDisplayName"; // Default display name
+        int listenerport = 25000; // Default listener port
+        std::string ipaddress = "127.0.0.1"; // Default IP address
+    };
+
+    struct CommunicationSettings {
+        int blocking = 0; // Default blocking setting
+        int socket_timeout = 5; // Default socket timeout
+    };
+
+    struct Logging {
+        std::string filename = "serverlog.txt"; // Default log file name
+        int log_level = 2; // Default log level
+        int flush = 0; // Default flush setting
+    };
+
+    struct Time {
+        int period_time = 30; // Default period time
+    };
+
+    struct ThreadPool {
+        int max_working_threads = 10; // Default max number of working threads
+    };
+
+    // Constructor that initializes the configuration from a file
+    Config(const std::string &filename) {
+        std::ifstream file(filename);
+        if (!file.is_open()) {
+            // If file cannot be opened, use default values
+            std::cerr << "Warning: Could not open filename. Settings are set to default values" << std::endl;
+        } else {
+            std::stringstream buffer;
+            buffer << file.rdbuf(); // Read the file content into a stringstream
+            JSON json = JSON::Parse(buffer.str()); // Parse the JSON content
+
+            ParseJSON(json); // Parse and set the configuration values
+        }
+    }
+
+    // Getters for configuration settings
+    Server get_Server() const { return server; }
+    CommunicationSettings get_CommunicationSettings() const { return communication_settings; }
+    Logging get_Logging() const { return logging; }
+    Time get_Time() const { return time; }
+    ThreadPool get_ThreadPool() const { return thread_pool; }
+
+private:
+    // Configuration members
+    Server server;
+    CommunicationSettings communication_settings;
+    Logging logging;
+    Time time;
+    ThreadPool thread_pool;
+
+    // Method to parse JSON and set configuration values
+    void ParseJSON(const JSON &json) {

```

Comment:
Should break down the ParseJSON method into smaller,  more focused sub-methods for better readability.